crates/bifrost/src/providers/replicated_loglet/network.rs

```diff
@@ -11,27 +11,40 @@
 // todo(asoli): remove when fleshed out
 #![allow(dead_code)]
 
+use std::cmp::Ordering;
 use std::pin::Pin;
 use std::sync::Arc;
 
+use dashmap::{DashMap, Entry};
 use futures::{Stream, StreamExt};
+use tracing::trace;
 
-use restate_core::network::{Incoming, MessageRouterBuilder, TransportConnect};
-use restate_core::{cancellation_watcher, Metadata};
+use restate_core::network::{Incoming, MessageRouterBuilder, Reciprocal, TransportConnect};
+use restate_core::{cancellation_watcher, task_center, Metadata, TaskKind};
 use restate_types::config::ReplicatedLogletOptions;
-use restate_types::net::replicated_loglet::Append;
-use tracing::trace;
+use restate_types::logs::{LogletOffset, SequenceNumber};
+use restate_types::net::replicated_loglet::{
+    Append, Appended, CommonRequestHeader, CommonResponseHeader, SequencerStatus,
+};
+use restate_types::replicated_loglet::ReplicatedLogletId;
 
+use super::loglet::ReplicatedLoglet;
 use super::provider::ReplicatedLogletProvider;
+use crate::loglet::util::TailOffsetWatch;
+use crate::loglet::{AppendError, LogletCommit};
 
 type MessageStream<T> = Pin<Box<dyn Stream<Item = Incoming<T>> + Send + Sync + 'static>>;
 
-pub struct RequestPump {
+pub struct RequestPump<T> {
     metadata: Metadata,
     append_stream: MessageStream<Append>,
+    active_loglet: DashMap<ReplicatedLogletId, Arc<ReplicatedLoglet<T>>>,
```

Comment:
The reason I did this, is that I need to get the Loglet Params from the chain. So I thought we could then cache it with the loglet id. 

Same reason I get the `last` segment of the logs on `get` otherwise we will have to scan the chain everytime to land on the correct segment index before i can fetch the params.

Reply:
Don't we send the index and log-id in the append message header?